src/composables/transaction.ts

```diff
@@ -99,18 +145,43 @@ export const useTransactionToken = (
     || getDexContracts.value.wae.includes(transaction.value.tx.contractId)
   ));
 
+  async function fetchOwnershipAccount(externalOwnerAddress: string | undefined) {
+    switch (ownershipStatus.value) {
+      case TRANSACTION_OWNERSHIP_STATUS.current:
+        return {
+          ...account.value,
+          label: i18n.t('transaction.overview.accountAddress'),
+          url: getExplorerPath.value(account.value.address),
+        };
+      case TRANSACTION_OWNERSHIP_STATUS.subAccount:
+        return accounts.value.find(
+          (acc) => [
+            transaction.value?.tx.accountId,
+            transaction.value?.tx.callerId,
+          ].includes(acc.address),
+        );
+      default: {
+        const address = externalOwnerAddress || txOwnerAddress.value;
+        const response = await fetchJson(`${activeNetwork.value.backendUrl}/profile/${address}`).catch(() => {});
```

Comment:
I think we should reuse existing logic and use `getPreferred` getter from `names` plugin. Which is done to other types onf transaction in `TransactionOverview` component.